includes/class-amp-theme-support.php

```diff
@@ -2025,11 +2059,14 @@ public static function prepare_response( $response, $args = [] ) {
 		 * See <https://www.w3.org/International/questions/qa-html-encoding-declarations>.
 		 */
 		if ( ! preg_match( '#<meta[^>]+charset=#i', substr( $response, 0, 1024 ) ) ) {
+			$meta_charset = sprintf( '<meta charset="%s">', esc_attr( get_bloginfo( 'charset' ) ) );
+
 			$response = preg_replace(
-				'/(<head[^>]*>)/i',
-				'$1' . sprintf( '<meta charset="%s">', esc_attr( get_bloginfo( 'charset' ) ) ),
+				'/(<head.*?>)/is',
```

Comment:
As the filtering regexp up above will let `<header>` elements count as well, this will also happily append the meta charset to the first `<header>` element in that case.